service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/BaseDevice.java

```diff
@@ -355,8 +350,6 @@ public void setForceUpdate(Boolean forceUpdate) throws IllegalArgumentException
         {
             throw new IllegalArgumentException();
         }
-
-        this.forceUpdate = forceUpdate;
```

Comment:
q - setForceUpdate now doesn't do anything other than throw an exception is the passed arg is null. Couldn't we just remove this method altogether?

Reply:
or is validation its only purpose? in that case, since we cannot rename a public method, we should probably add a comment explaining that.

Reply:
In this context, addressing the warning for forceUpdate which is not referenced/used anywhere else. For the implementation. I did my research but haven't found why this was implemented in the first place!! I will address adding comment in follow up PR.